### DISCUSSION TITLE
Clarification on TODO item: "Display source code from markdown more correctly."

### DISCUSSION THREAD

**Discussion by varphi-online** [TimoKats/mdrss#25] [General]
Hiya,

I would love to try to start implementing this TODO item, but would like some clarification on the phrase "more correctly". I interpret this as possibly enclosing .md code blocks in the html \<code> tag, or possibly using something like the [Golang Chroma](https://pkg.go.dev/github.com/alecthomas/chroma) library to apply inline styling, per-language, with a config option, but don't want to work on something that isn't needed, thank you!

**Comment by TimoKats** [comment]
Hey Varphi, thanks for picking up another issue! Good question. That TODO point refers to showing markdown indicators (e.g. \```python) in the RSS feed. There are a couple ways we can solve that, but we have to keep some limitations in mind.

- Most people (incl me) use newsboat (or another terminal based application) to get their feeds. As a result, color styling probably won't come across. Moreover, adding more complex styling is also adding more things that can break.
- People have different go versions on their systems. So adding external dependencies could break user spaces.

As a result, I'm in favor of the simpler approach. The `<code>` block seems like a good place to start :+1:  I'm also curious to test how newsboat deals with that.

**Comment by TimoKats** [comment]
@varphi-online I quite like the way it was solved btw, including the new case switch statement. Good work :1st_place_medal: 

As for next steps, I think I will plan a broader refactoring somewhere in the future. I'm busy with some other projects now (that I plan to release soon). But, if you can think of any lacking features feel free to open a PR/Issue whenever.

PS: Your coding is quite good, especially considering you're new to GO. So feel free to contribute to anything you find on my github. Also, if I can ever help you out (with something coding related), please let me know. Preferably using the email available on my website (see github profile). I owe you one for these contributions :+1:
